src/Truncate.js

```diff
@@ -10,14 +10,16 @@ export default class Truncate extends Component {
             PropTypes.number
         ]),
         trimWhitespace: PropTypes.bool,
-        onTruncate: PropTypes.func
+        onTruncate: PropTypes.func,
+        width: PropTypes.number
     };
 
     static defaultProps = {
         children: '',
         ellipsis: '…',
         lines: 1,
-        trimWhitespace: false
+        trimWhitespace: false,
+        width: 0
```

Comment:
Could we change this to something like `shouldUpdate` or `forceReRerender` (and make it boolean) since this is the what we want at the end. It would be also nice if we have at least one test for the case when explicitly passing `true`.

Reply:
@moubi That won't work, since you wouldn't get notified via `componentDidUpdate` if you simply passed `true`.

To force recalculation imperatively, you can do:

```js
···
recalculate() {
  this.truncate.onResize();
}

render() {
  return (
    <Truncate refs={ref => this.truncate = ref}>···</Truncate>
  );
}
···
```

I think this PR is actually the most elegant solution, if you want to solve the problem by using props.